cpp/independent_set/ExactRecursiveIndependentSet.cpp

```diff
@@ -0,0 +1,755 @@
+                graph->restoreNode(v);
+                graph->addEdge(v, u1);
+                graph->addEdge(v, u2);
+                graph->addEdge(v, u3);
+
+                std::vector<NetworKit::node> delVerticesCase1 = vNeighborsPlus;
+                std::vector<NetworKit::node> delVerticesCase2 = {v};
+                delVerticesCase2.insert(delVerticesCase2.end(), u1NeighborsPlus.begin(), u1NeighborsPlus.end());
+                delVerticesCase2.insert(delVerticesCase2.end(), u2NeighborsPlus.begin(), u2NeighborsPlus.end());
+                std::vector<NetworKit::node> delVerticesCase3 = {v, u2};
+                delVerticesCase3.insert(delVerticesCase3.end(), u1NeighborsPlus.begin(), u1NeighborsPlus.end());
+                delVerticesCase3.insert(delVerticesCase3.end(), u3NeighborsPlus.begin(), u3NeighborsPlus.end());
+                std::vector<NetworKit::node> delVerticesCase4 = {v, u1};
+                delVerticesCase4.insert(delVerticesCase4.end(), u2NeighborsPlus.begin(), u2NeighborsPlus.end());
+                delVerticesCase4.insert(delVerticesCase4.end(), u3NeighborsPlus.begin(), u3NeighborsPlus.end());
+
+                EdgeSet connectedEdgesCase1 = getConnectedEdges(delVerticesCase1);
+                EdgeSet connectedEdgesCase2 = getConnectedEdges(delVerticesCase2);
+                EdgeSet connectedEdgesCase3 = getConnectedEdges(delVerticesCase3);
+                EdgeSet connectedEdgesCase4 = getConnectedEdges(delVerticesCase4);
+
+                std::vector<NetworKit::node> setCase1, setCase2, setCase3, setCase4;
+                removeElements(delVerticesCase1);
+                setCase1 = recursive();
+                setCase1.push_back(v);
+                restoreElements(delVerticesCase1, connectedEdgesCase1);
+
+                removeElements(delVerticesCase2);
+                setCase2 = recursive();
+                setCase2.push_back(u1);
+                setCase2.push_back(u2);
+                restoreElements(delVerticesCase2, connectedEdgesCase2);
+
+                removeElements(delVerticesCase3);                
+                setCase3 = recursive();
+                setCase3.push_back(u1);
+                setCase3.push_back(u3);
+                restoreElements(delVerticesCase3, connectedEdgesCase3);
+
+                removeElements(delVerticesCase4);
+                setCase4 = recursive();
+                setCase4.push_back(u2);
+                setCase4.push_back(u3);
+                restoreElements(delVerticesCase4, connectedEdgesCase4);
+
+                if (setCase1.size() >= setCase2.size() && setCase1.size() >= setCase3.size() && setCase1.size() >= setCase4.size()) return setCase1;
+                if (setCase2.size() >= setCase1.size() && setCase2.size() >= setCase3.size() && setCase2.size() >= setCase4.size()) return setCase2;
+                if (setCase3.size() >= setCase1.size() && setCase3.size() >= setCase2.size() && setCase3.size() >= setCase4.size()) return setCase3;
```

Comment:
Consider refactoring: create vector "solutions", put there all branches + apply std::min_element to pick the best one